### PR TITLE
Arc-2302 update logging

### PR DESCRIPTION
**Pull request by rachellerathbone**
**What's in this PR?**
Updated the logs with a temporary logger that uses console log.

**Why**
Forge currently doesn't support custom loggers (only supports console logs :sigh: so you wind up with a bunch of errors. Apparently they're working on it though. Also double checked what bunyan streams by default and it's process.stdout (which causes the errors in Forge apps).

In part 1 I've just added a new logger and replaced the existing 'analytics logger'

**Added feature flags**
no can do atm but ff support coming soon...

**Affected issues**  
Arc-2302

**How has this been tested?**  
Locally:

Screenshot 2023-07-31 at 5 01 48 pm

Screenshot 2023-07-31 at 5 06 43 pm

Screenshot 2023-07-31 at 5 06 13 pm

Screenshot 2023-07-31 at 5 04 36 pm

**What's Next?**
Part 2 - replacing console logs

### CODE REVIEW

File: app/jenkins-for-jira-ui/src/common/analytics/analytics-client.ts [hunk 1]
```diff
@@ -21,13 +22,6 @@ interface TenantType {
 	TRELLO_WORKSPACE_ID: string
 }
 
-enum EnvType {
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/common/analytics/analytics-client.ts:24]: Moved this out into a separate file to emulate the same behaviour on front/back ends

File: app/src/common/types.ts [hunk 2]
```diff
@@ -20,3 +20,11 @@ export interface JenkinsPipeline {
 	lastEventStatus: BuildEventStatus | DeploymentEventStatus,
 	lastEventType: EventType
 }
+
+export enum EnvironmentEnum {
```
Comment (rachellerathbone) [inline app/src/common/types.ts:24]: Type for backend. Can't be imported into the client.

File: app/src/config/logger.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	logLevel: keyof typeof LogColors,
+	name: string,
+	timestamp: string,
+	formattedLogData: string
+): string => {
+	const logColor = LogColors[logLevel];
+	return `${logColor}${logLevel}${LogColors.RESET}: [${timestamp}] ${name} - ${formattedLogData}`;
+};
+
+const log = (
```
Comment (rachellerathbone) [inline app/src/config/logger.ts:32]: Definitely not the best but better than what we had 🤷

File: app/src/storage/connect-jenkins-server.ts [hunk 4]
```diff
@@ -1,21 +1,45 @@
...
-		await storage.setSecret(`${SECRET_STORAGE_KEY_PREFIX}${jenkinsServer.uuid}`, secret);
-		console.log(`${jenkinsServer.name} jenkins server configuration saved successfully !`);
-		log({ eventType: 'jenkinsServerCreated', data: { uuid: jenkinsServer.uuid } });
+		const { uuid } = jenkinsServer;
+		await storage.set(`${SERVER_STORAGE_KEY_PREFIX}${uuid}`, jenkinsServer);
+		await storage.setSecret(`${SECRET_STORAGE_KEY_PREFIX}${uuid}`, secret);
+
+		log(
+			logName,
+			'INFO',
```
[the remark this answers is not in the document]
Comment (bgvozdev) [reply app/src/storage/connect-jenkins-server.ts:20]: shall we maybe add aliases (`logInfo`, `logWarn`) etc?

**Pull request closed without merging**